src/plugins/assetUploader/index.ts

```diff
@@ -0,0 +1,78 @@
+import { AptosConfig } from "../../api/aptosConfig";
+import { Account } from "../../core";
+import { IrysAssetUploader } from "./irys";
+import { FundResponse, IAssetUploader, UploadResponse } from "./type";
+
+export class AssetUploader {
+  readonly config: AptosConfig;
+
+  readonly assetUploader: IAssetUploader;
+
+  private constructor(config: AptosConfig, assetUploader: IAssetUploader) {
+    this.config = config;
+    this.assetUploader = assetUploader;
+  }
+
+  static async init(config: AptosConfig) {
+    switch (config.assetUploaderProvider) {
+      case "irys": {
+        const assetUploader = await IrysAssetUploader.init(config);
+        return new AssetUploader(config, assetUploader);
+      }
+      default:
+        throw new Error("Invalid asset uploader provided");
+    }
+  }
+
```

Comment:
we have a separate `init` func cuz irys is optional?

Reply:
no, cuz we use async/await to [detect whether the irys module exists](https://github.com/aptos-labs/aptos-ts-sdk/pull/318/files#diff-9601dd624fda4574fcf57325ac1ddadbad6fc6ba5337b5735161b369e989493aR24) - since a constructor can not be an async function, we need to have a `static async` function and a private constructor.